src/CycloneDX.Core/Models/Component.cs

```diff
@@ -178,15 +178,18 @@ public bool NonNullableModified
         public List<ExternalReference> ExternalReferences { get; set; }
         public bool ShouldSerializeExternalReferences() { return ExternalReferences?.Count > 0; }
 
-        [XmlArray("components")]
-        [ProtoMember(21)]
-        public List<Component> Components { get; set; }
+        //In the xml format, Properties is in front of Components.
+        //XML serialization uses the member order unless explicitly specified differently.
 
         [XmlArray("properties")]
         [XmlArrayItem("property")]
         [ProtoMember(22)]
         public List<Property> Properties { get; set; }
         public bool ShouldSerializeProperties() { return Properties?.Count > 0; }
+
+        [XmlArray("components")]
+        [ProtoMember(21)]
```

Comment:
Just in case: `ProtoMember` numbers should remain as they were, right? Or should match the order in other serializers?..

Reply:
Yes, they need to remain as they are, they are defined in
https://github.com/CycloneDX/specification/blob/c320fc0f0b46873864927d9d5684eea7ba439728/schema/bom-1.4.proto#L110-L113
(The order in the proto is different from the one in the xml.)